pkg/azure/actuator.go

```diff
@@ -230,36 +230,56 @@ func (a *Actuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) er
 		logger.Debug("credentials already up to date")
 		return nil
 	}
-
-	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
+	stsDetected, err := utils.IsTimedTokenCluster(a.client, ctx, logger)
 	if err != nil {
-		logger.WithError(err).Error("issue with cloud credentials secret")
 		return err
 	}
-
-	switch credentialsRootSecret.Annotations[constants.AnnotationKey] {
-	case constants.InsufficientAnnotation:
-		msg := "cloud credentials insufficient to satisfy credentials request"
-		logger.Error(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.InsufficientCloudCredentials,
-			Message:   msg,
+	logger.Infof("stsDetected: %v", stsDetected)
+	if stsDetected {
+		logger.Debug("actuator detected Azure AD Workload Identity enabled cluster, enabling Workload Identity secret brokering for CredentialsRequests providing a Managed Identity")
+		azureProviderSpec, err := decodeProviderSpec(minterv1.Codec, cr)
+		if err != nil {
+			return err
+		}
+		azureFederatedTokenFile := cr.Spec.CloudTokenPath
+		if cr.Spec.CloudTokenPath == "" {
+			logger.Debug("CredentialsRequest has no cloudTokenPath, defaulting azure_federated_token_file to /var/run/secrets/kubernetes.io/serviceaccount/token")
+			azureFederatedTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
 		}
-	case constants.PassthroughAnnotation:
-		logger.Debugf("provisioning with passthrough")
-		err := a.syncPassthrough(ctx, cr, credentialsRootSecret, logger)
+		err = a.syncWorkloadIdentitySecret(*azureProviderSpec, azureFederatedTokenFile, cr, logger, ctx)
 		if err != nil {
 			return err
 		}
-	default:
-		msg := fmt.Sprintf("unexpected value or missing %s annotation on admin credentials Secret", constants.AnnotationKey)
-		logger.Info(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.CredentialsProvisionFailure,
-			Message:   msg,
+	} else {
+		credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
```

Comment:
This can be flattened...
```diff
-		if err != nil {
-			return err
-		}
+		return err
-	} else {
-		credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
+	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
<truncated>
```